src/utils/customHTML.ts

```diff
@@ -0,0 +1,109 @@
+}
+
+/**
+ * Retrieves the custom HTML tag for embedding an Instagram post
+ * @param postId the ID of the Instagram post
+ * @param isCaptioned whether to include the post caption
+ * @returns the Instagram post custom HTML tag code
+ */
+export const getInstagramEmbedTag = (
+  postId: string | undefined,
+  isCaptioned: boolean
+): string => {
+  const extraAttr = isCaptioned ? " captioned" : ""
+
+  if (!postId) {
+    return ""
+  }
+
+  return `<instagram-post post-id="${postId}" ${extraAttr}></instagram-post>`
+}
+
+/**
+ * Processes the given HTML chunk by converting the custom Instagram tag
+ * into the appropriate HTML for embedding the Instagram post.
+ * Refer to getInstagramEmbedTag() for the custom tag format.
+ * @param html the HTML chunk to process
+ * @returns the processed HTML chunk
+ */
+export const processInstagramTagToEmbed = (html: string): string => {
+  const $ = cheerio.load(html, null, false)
+  $("instagram-post").each((i: number, el: Element) => {
+    const postId = $(el).attr("post-id")
+    const isCaptioned = !!(
+      ($(el).attr("captioned") || $(el).attr("captioned") === "") &&
+      $(el).attr("captioned") !== "false"
+    )
+    const embedHtml = getInstagramEmbedCode(postId, isCaptioned)
+    $(el).replaceWith(embedHtml)
+  })
+
+  return $.html()
+}
+
+/**
+ * Processes the given HTML chunk by converting the raw HTML code for
+ * embedding an Instagram post into the custom HTML tag format suitable
+ * for the Markdown editor.
+ * Refer to getInstagramEmbedCode() for the raw HTML code format.
+ * @param html the HTML chunk to process
+ * @returns the processed HTML chunk
+ */
+export const processInstagramEmbedToTag = (html: string): string => {
+  const $ = cheerio.load(html, null, false)
+  $("blockquote.instagram-media").each((i: number, el: Element) => {
+    const postUrl = $(el).attr("data-instgrm-permalink")
+    const postId = postUrl?.split("/p/")[1].split("/")[0]
+    const isCaptioned = !!(
+      ($(el).attr("data-instgrm-captioned") ||
+        $(el).attr("data-instgrm-captioned") === "") &&
+      $(el).attr("data-instgrm-captioned") !== "false"
```

Comment:
Hmm this condition seems off..  if `$(el).attr("data-instgrm-captioned") === ""` is true, then `$(el).attr("data-instgrm-captioned") !== "false` is definately true

Reply:
Updated this condition and added a comment, thanks!